include/fluent-bit/flb_task.h

```diff
@@ -88,6 +106,120 @@ struct flb_task {
     struct flb_config *config;           /* parent flb config             */
 };
 
+static inline int flb_task_route_lifecycle_hook_add(struct flb_task *task,
+                                             int hook_type,
+                                             int durability,
+                                             void *id,
+                                             struct flb_output_instance *out_instance,
+                                             void (*callback_ptr)(void *),
+                                             void *data)
+{
+    struct mk_list *tmp;
+    struct mk_list *head;
+    struct flb_task_route *route;
```

Comment:
for pointers, initialize them to NULL to avoid dangling pointers used in the code.